python/main.py

```diff
@@ -22,21 +25,75 @@
 def root():
     return {"message": "Hello, world!"}
 
-@app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+@app.get("/items")
+def get_items():
+    try:
+        with open("items.json", "r") as f:
+            data = json.load(f)
+            items = data.get("items", [])
+        return {"items": items}
+    except:
+        return {"message": "Items not registered"}
+
+@app.get("/items/{item_id}")
+def get_item(item_id: int):
+    try:
+        with open("items.json", "r") as f:
+            data = json.load(f)
+            items = data.get("items", [])
+            found_items = [item for item in items if item.get("id") == item_id]
+            if found_items:
+                item = found_items[0]
+                return {
+                    "name": item.get("name"),
+                    "category": item.get("category"),
+                    "image_filename": item.get("image_filename")
+                }
+            else:
+                return {
+                    {"message": "Item not found"}
+                }
+    except:
+        return {"message": "Items not registered"}
 
 @app.get("/image/{image_filename}")
-async def get_image(image_filename):
+def get_image(image_filename):
     # Create image path
     image = images / image_filename
 
     if not image_filename.endswith(".jpg"):
         raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
-
     if not image.exists():
         logger.debug(f"Image not found: {image}")
         image = images / "default.jpg"
 
-    return FileResponse(image)
\ No newline at end of file
+    return FileResponse(image)
+
+@app.post("/items")
+async def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
+    logger.info(f"Receive item: {name} category: {category} image: {image.filename}")
+    # Save image
+    content_image = await image.read()
+    hash_object = hashlib.sha256(content_image)
+    image_filename = hash_object.hexdigest() + ".jpg"
+    path_image_file = "images/"+ image_filename
+    with open(path_image_file , "wb") as f:
+        f.write(content_image)
+
+    # Save in json
+    new_item = {"name": name, "category": category, "image_filename": image_filename}
+    try:
+        with open("items.json", "r") as f:
+            data = json.load(f)
+            items = data.get("items", [])
+            new_item["id"] = len(items)
+            items.append(new_item)
+    #If "items.json" file doesn't exist, items load only new_item
+    except FileNotFoundError:
+        new_item["id"] = 0
+        items = [new_item]
+
+    with open("items.json", "w") as f:
+        json.dump({"items": items}, f)
+
+    return {"message": f"received item:{name} category:{category}"}
```

Comment:
[nit][imo] このままでもいいですが、セキュリティ的な意味を考えるとここで返すメッセージは`{"message": "success"}`などだけにして、`{"message": f"received item:{name} category:{category}"}`はloggerで内容を出す方がいいかもしれません。

```suggestion
    logger.dubug(f"received item: {name} category: {category}")
    return {"message": f"succeed to add item"}
```